cpp/src/strings/convert/convert_hex.cu

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/strings/convert/convert_integers.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <cudf/utilities/traits.hpp>
+#include <strings/utilities.hpp>
+#include <strings/utilities.cuh>
+
+#include <rmm/thrust_rmm_allocator.h>
+#include <thrust/transform.h>
+#include <thrust/iterator/counting_iterator.h>
+
+namespace cudf
+{
+namespace strings
+{
+namespace detail
+{
+namespace
+{
+
+/**
+ * @brief Converts hex strings into an integers.
+ *
+ * Used by the dispatch method to convert to different integer types.
+ */
+template <typename IntegerType>
+struct hex_to_integer_fn
+{
+    column_device_view const strings_column;
+
+    /**
+     * @brief Converts a single hex string into an integer.
+     *
+     * Non-hexadecimal characters are ignored.
+     * This means it can handle "0x01A23" and "1a23".
+     *
+     * Overflow of the int64 type is not detected.
+     */
+    __device__ int64_t string_to_integer( string_view const& d_str )
+    {
+        int64_t result = 0, base = 1;
+        const char* str = d_str.data();
+        size_type index = d_str.size_bytes();
+        while( index-- > 0 )
+        {
+            char ch = str[index];
+            if( ch >= '0' && ch <= '9' )
+            {
+                result += static_cast<int64_t>(ch-48) * base;
+                base <<= 4; // multiply by 16
```

Comment:
The compiler is smart enough to do this optimization for you and it's easier to read this way :) 
```suggestion
                base *= 16; // multiply by 16
```